src/database-layer/db.ts

```diff
@@ -555,21 +562,28 @@ if (maybeMysql != null) {
 			}
 		}
 
-		return _.extend(
-			{
-				engine: Engines.mysql,
-				executeSql: atomicExecuteSql,
-				transaction: createTransaction(stackTraceErr =>
-					connect().then(client => {
-						const close = () => client.release();
-						const tx = new MySqlTx(client, close, stackTraceErr);
-						tx.executeSql('START TRANSACTION;');
-						return tx;
-					}),
-				),
-			},
-			alwaysExport,
-		);
+		return {
+			engine: Engines.mysql,
+			executeSql: atomicExecuteSql,
+			transaction: createTransaction(stackTraceErr =>
+				connect().then(client => {
+					const close = () => client.release();
+					const tx = new MySqlTx(client, close, stackTraceErr);
+					tx.executeSql('START TRANSACTION;');
+					return tx;
+				}),
+			),
+			readTransaction: createTransaction(stackTraceErr =>
+				connect().then(client => {
+					const close = () => client.release();
+					const tx = new MySqlTx(client, close, stackTraceErr);
+					tx.executeSql('SET TRANSACTION READ ONLY;');
+					tx.executeSql('START TRANSACTION;');
```

Comment:
These statements are in the opposite order of that in Postgres, is this deliberate?

Reply:
Yeah, in postgres the change applies to the current transaction (and so is essentially a no-op outside of a transaction), in mysql it applies to the next transaction and will throw an error if run inside of one - just fun little differences :P

Reply:
Cool, that's what I thought as well but didn't bother to lookup docs and wanted to confirm this isn't by accident 👍